sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdInfoLiveData.java

```diff
@@ -0,0 +1,477 @@
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Transformations;
+import androidx.work.Data;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+import java.util.List;
+
+/**
+ * Package private.
+ *
+ * A type to create the LiveData pair, a {@link TransferIdOrError} LiveData and associated {@link TransferInfo}
+ * LiveData. Each transfer will have such a unique LiveData pair.
+ *
+ * When transfer id is set to TransferIdOrError LiveData then TransferInfo LiveData streams {@link TransferInfo}
+ * events describing current state of the transfer identified by the transfer id. Internally the source of the
+ * TransferInfo LiveData queries {@link WorkManager} for a LiveData that streams {@link WorkInfo} and transform
+ * {@link WorkInfo} events to {@link TransferInfo} events.
+ *
+ * Same transfer id can be set multiple times in the TransferIdOrError LiveData. Each such set results in querying
+ * {@link WorkManager} as described above. There can be active Observes for an original transfer when it is paused
+ * and later resumed by a different Worker. When possible we share the same LiveData pair between original transfer
+ * and it's resume. In such cases, the ability to set transfer id on the TransferIdOrError LiveData multiple times
+ * enables to transparently switch the Worker and keep any existing Observers of the original transfer to continue
+ * to receive the {@link TransferInfo} events.
+ *
+ * @see TransferIdInfoLiveDataCache (for LiveData pair sharing)
+ */
+final class TransferIdInfoLiveData {
+    private static final String TAG = TransferIdInfoLiveData.class.getSimpleName();
+    // the input LiveData that receives the transfer id or error.
+    private final MutableLiveData<TransferIdOrError> transferIdOrErrorLiveData = new MutableLiveData<>();
+    // the output TransferInfo LiveData.
+    private final MediatorLiveData<TransferInfo> transferInfoLiveData = new MediatorLiveData<>();
+    // the the object that TransferClient methods update any state that it want TransferInfo LiveData source
+    // for a transfer to know.
+    private final TransferFlags transferFlags = new TransferFlags();
+    // the recent TransferIdOrError object emitted by the transferIdOrErrorLiveData object.
+    private TransferIdOrError inputTransferIdOrError;
+    // hold the state of the last WorkInfo received from Transfer Worker.
+    private WorkInfo.State lastWorkInfoState;
+    // flag to track whether current event about to emit from transferInfoLiveData is the first event.
+    private boolean isFirstEvent = true;
+    // flag to track whether the last emitted event from transferInfoLiveData was a pause event.
+    private boolean wasPaused;
+
+    private TransferIdInfoLiveData() {}
+
+    @MainThread
+    static TransferIdInfoLiveData.Result create(Context context) {
+        TransferIdInfoLiveData transferIdInfoLiveData = new TransferIdInfoLiveData();
```

Comment:
This may be a dumb question but why are we using a static method to create this instead of making `init()` not private and using it?

Reply:
Mainly for readability, the name `create` indicates it is factory method hence is static but `init` is initializing an already created object.